### PR TITLE
[Pagination] Phase III - Paginator Logic & Finalize Pagination Cache Lifecycle

### PR DESCRIPTION
**Pull request by quaark**
https://iguazio.atlassian.net/browse/ML-5803

Implement paginator wrapper logic that wraps paginated functions with handling the cached record CRUD.

When requesting to paginate a request:
- If the method isn't supported raise NotImplemented
- If no page size or token is given, return all items - no pagination
- If token was given, check db cache records
  - Record doesn't exist? - 404
  - Record belongs to other user? - 403
  - Update record's last_access time and current page
- Call supported method with page number, page size
  - method raised StopIteration? - Clear cache record from db
- return response with pagination info (token, page number, page size)

### CODE REVIEW

**Review by liranbg — APPROVED** [review]

File: server/api/utils/pagination.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import server.api.crud
+from mlrun.utils import logger
+
+
+class PaginatedMethods:
+    _methods: list[typing.Callable] = [
+        # TODO: add methods when they implement pagination
+        # server.api.crud.Runs().list_runs,
+    ]
+    _method_map = {method.__name__: method for method in _methods}
```
Comment (theSaarco) [inline server/api/utils/pagination.py:31]: If you implemented `get_name`, why not use it here?

File: tests/api/utils/test_pagination.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        paginator.paginate_request(
+            db, paginated_method, auth_info_2, pagination_info["token"]
+        )
+
+    logger.info(
+        "Requesting second page without auth info, should raise AccessDeniedError"
+    )
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        paginator.paginate_request(db, paginated_method, None, pagination_info["token"])
```
Comment (theSaarco) [inline tests/api/utils/test_pagination.py:211]: What about cleaning up the pagination cache? Aren't you concerned about pagination cache leaks across tests?

**Review by theSaarco — COMMENTED** [review]
Nice work. Have some small comments/questions.

File: tests/api/utils/test_pagination.py (hunk 2 above)
Comment (quaark) [reply tests/api/utils/test_pagination.py:211]: Not really as we delete the db when the test goes down. But can add it to the teardown

File: server/api/utils/pagination.py (hunk 1 above)
Comment (quaark) [reply server/api/utils/pagination.py:31]: Sure 👍

Comment (quaark) [reply server/api/utils/pagination.py:31]: Spoke offline removed get_name

**Pull request merged**